### PR TITLE
Tax gui and backend without geolocation #64 #63 #62 #51

### PR DESCRIPTION
**Pull request by almiche**
### WHAT kind of change does this PR introduce?
Added the tax calculation feature minus geolocation which is being done by @miisaka 
The following is a demo of the feature in action.
gif of tax feature
...

### HOW is this accomplished?
By adding a dialog which allows the user to select their location.
...

### Checklist
- [ ] Docs have been added / updated to reflect the changes
- [x] I have reviewed and tested the changes :white_check_mark:
- [x] I have squashed my commits to have a reasonable amount of commits
- [x] All of the code I have written adhere to the coding conventions outlined in the coding conventions section of the wiki

### CODE REVIEW

**Comment by izconcept** [comment]
Are we covering the case where the user gets geolocated to somewhere not in Canada? What happens then? I'm assuming @miisaka will implement that?

**Comment by almiche** [comment]
@izconcept  we are only doing provinces in canada.

**Comment by miisaka** [comment]
Looks good, will need to implement the actual Geolocation service though. I will do that once you merge this PR. Great work

File: app/src/main/kotlin/com/simplemobiletools/calculator/helpers/MoneyCalculatorImpl.kt [hunk 1]
```diff
@@ -84,4 +85,17 @@ class MoneyCalculatorImpl(calculator: Calculator, val context: Context) {
...
+        //Code to ping location here
+        if(false){
+
+        }else{
+            tCallback!!.spawnTaxModal()
+        }
+    }
+
+    fun performTaxing(location:String){
+        overwriteNumber(TaxOperation(Formatter.stringToDouble(mCallback!!.getResult()),location).getResult())
```
[the remark this answers is not in the document]
Comment (almiche) [reply app/src/main/kotlin/com/simplemobiletools/calculator/helpers/MoneyCalculatorImpl.kt]: Done

**Comment by izconcept** [comment]
**Styling Review:** PR conforms to our established style guidelines. Good Styling!

**Pull request merged**